wdl-lint/src/rules/shellcheck.rs

```diff
@@ -396,6 +398,39 @@ fn to_bash_var(placeholder: &Placeholder, ty: Option<Type>) -> (String, bool) {
                     true,
                 );
             }
+            PrimitiveType::String => {
+                match placeholder.expr() {
+                    Expr::If(i) =>  {
+                        let (_, if_expr, else_expr) = i.exprs();
+                        if let (
+                            Expr::Literal(LiteralExpr::String(_)),
+                            Expr::Literal(LiteralExpr::String(_)),
+                        ) = (if_expr, else_expr)
+                        {
+                            return ("a".repeat(placeholder_len), true);
+                        }
+                    }
+                    Expr::Parenthesized(p) => {
+                        if let Expr::If(i) = p.expr() {
+                            let (_, if_expr, else_expr) = i.exprs();
+                            if let (
+                                Expr::Literal(LiteralExpr::String(_)),
+                                Expr::Literal(LiteralExpr::String(_)),
+                            ) = (if_expr, else_expr)
+                            {
+                                return ("a".repeat(placeholder_len), true);
+                            }
+                        }
+                    }
+                    Expr::Call(c) => {
+                        let target = c.target();
+                        if target.text() == "sep" {
+                            return ("a".repeat(placeholder_len), true);
+                        }
+                    }
```

Comment:
This should probably get an actual lint rule to suggest that the array argument to `sep` should have a `quote()` or `squote()` function surrounding it.